dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Helper/ProviderService.cs

```diff
@@ -1,29 +1,152 @@
-﻿using System.Collections.Generic;
-using System.Net.Http;
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
 using System.Threading.Tasks;
+using Dapper;
 using Dfc.Providerportal.FindAnApprenticeship.Interfaces.Services;
 using Dfc.Providerportal.FindAnApprenticeship.Models.Providers;
-using Newtonsoft.Json;
 
 namespace Dfc.Providerportal.FindAnApprenticeship.Helper
 {
     public class ProviderService : IProviderService
     {
-        private readonly HttpClient _httpClient;
+        private readonly Func<IDbConnection> _dbConnectionFactory;
 
-        public ProviderService(HttpClient httpClient)
+        public ProviderService(Func<IDbConnection> dbConnectionFactory)
         {
-            _httpClient = httpClient;
+            _dbConnectionFactory = dbConnectionFactory ?? throw new ArgumentNullException(nameof(dbConnectionFactory));
         }
 
         public async Task<IEnumerable<Provider>> GetActiveProvidersAsync()
         {
-            var response = await _httpClient.GetAsync($"GetActiveProviders");
+            const string sql = @"
+                select * from Pttcd.ProviderContacts pc
```

Comment:
Maybe consider removing leading spaces here as the spaces end up in the logs too